src/validators/SessionKeyValidator.sol

```diff
@@ -68,30 +68,31 @@ contract SessionKeyValidator is IValidationHook, IModuleValidator {
 
   function init(bytes calldata data) external {
     // to prevent duplicate inits, since this can be hook plus a validator
-    if (!_isInitialized(msg.sender) && data.length != 0) {
+    if (!_isHookAndModuleInitialized(msg.sender) && data.length != 0) {
       require(_addValidationKey(data), "init failed");
     }
   }
 
   function _addValidationKey(bytes memory sessionData) internal returns (bool) {
-    if (sessionData.length == 0) {
-      return false;
-    }
     SessionLib.SessionSpec memory sessionSpec = abi.decode(sessionData, (SessionLib.SessionSpec));
     createSession(sessionSpec);
     return true;
   }
 
   function disable() external {
-    if (_isInitialized(msg.sender)) {
+    if (_isHookInitialized(msg.sender)) {
       // Here we have to revoke all keys, so that if the module
       // is installed again later, there will be no active sessions from the past.
       // Problem: if there are too many keys, this will run out of gas.
       // Solution: before uninstalling, require that all keys are revoked manually.
       require(sessionCounter[msg.sender] == 0, "Revoke all keys first");
```

Comment:
this `require` should probably be outside of the `if`, since a user can uninstall the module as validator and bypass this check